### PR TITLE
Pass AuthError in Realm.Sync.User methods callbacks

### PR DESCRIPTION
**Pull request by fealebenpae**
The new Realm.Sync.AuthError class exposes properties common to the Problem family of classes in the Realm Object Server

### CODE REVIEW

File: lib/extensions.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ module.exports = function(realmConstructor) {
 
     // Add sync methods
     if (realmConstructor.Sync) {
-        realmConstructor.Sync.User = require('./sync').User;
```
Comment (alazier) [inline lib/extensions.js]: Please remove this change as this conflicts with coming changes in the `al-sync-user` branch.

**Comment by alazier** [comment]
This looks fine. But we should be merging stuff into `al-sync`/standardizing on the main sync branch we are working off of.

**Comment by fealebenpae** [comment]
Which is the main sync branch? `al-sync` was last updated 20 days ago. Shouldn't we be merging to `master` by now?

File: lib/sync.js [hunk 2]
```diff
@@ -39,21 +41,19 @@ const postHeaders =  {
...
     json.app_id = '';
     var options = {
-        url: server + 'auth',
+        url: server + '/auth',
         body: JSON.stringify(json),
         headers: postHeaders
     };
     post(options, function(error, response, body) {
         if (error) {
-            console.log(error);
```
Comment (fealebenpae) [inline lib/sync.js]: IMO plainly `console.log`-ing errors in a library looks very unprofessional to me, but I can bring back the `console.log` statements if you like.

**Comment by alazier** [comment]
We can't merge to master until we have proper testing and we are ready to release sync beta for RN.

File: lib/sync.js (hunk 2 above)
Comment (alazier) [inline lib/sync.js]: I agree. That is why we should replace this with logging only when the appropriate log level is set.

**Comment by fealebenpae** [comment]
I understand. I'm happy to rebase this PR on another branch, just tell me which one I should use.

**Comment by alazier** [comment]
I guess you can merge into this branch, but we should change https://github.com/realm/realm-js/pull/609 to merge into `al-sync`. Once I finish the user persistence work I think we can get things merged to master.

**Comment by fealebenpae** [comment]
I changed #609 to be based off of `al-sync` and I'm merging this into #609's branch.

**Pull request merged**